feeds/tip/certificates/files/usr/bin/mount_certs

```
@@ -69,12 +67,15 @@ sonicfi,rap6*)
 			fi
 		fi
 
-		if mount | grep "/certificates" | grep -qE "squashfs|ubifs"; then
-			mkdir -p /tmp/certs_backup
-			cp -a /certificates/* /tmp/certs_backup/
-			umount /certificates
-			mkdir -p /certificates
-			cp -a /tmp/certs_backup/* /certificates
+		overlay_name="certs_overlay"
+
+		if mount | grep "/certificates" | grep -qE "squashfs|ubifs" && \
+			! mount | grep "/certificates" | grep -q "$overlay_name"; then
+			mkdir -p /tmp/certs_upper /tmp/certs_work /tmp/certs_merged
+			mount -t overlay "$overlay_name" \
```

why do we need an overlay_fs here ? there are 2 optiosn
1) it is ubifs, then we do not have to do anything
2) the certs are in a tar file and can be unpacked to the /certificates folder
both options do not require the complexity of using overlayfs

---

Thanks for the question. Here's why overlayfs is included:

- squashfs is read-only  
  During enrollment, new PKI 2.0 certificates are written to /certificates before being archived by store_certs.  
  If it's mounted as squashfs, these writes silently fail and won’t be included in the backup tar.  
  Overlayfs makes the path writable and avoids this issue.

- UBIFS is writable, but we kept the overlay logic for consistency  
  While UBIFS doesn't require overlayfs, maintaining a unified logic path avoids branching code based on filesystem type.  
  This simplifies the codebase and reduces future maintenance effort.